src/rdf4cpp/rdf/datatypes/registry/DatatypeRegistry.hpp

```diff
@@ -332,6 +332,15 @@ struct SelectOpRes<std::false_type, Fallback> {
     using type = Fallback;
 };
 
+template<typename T>
+[[nodiscard]] nonstd::expected<std::any, NumericOpError> map_expected(nonstd::expected<T, NumericOpError> const &e) noexcept {
+    if (e.has_value()) {
+        return *e;
+    } else {
+        return e.error();
```

Comment:
Is this safe? Couldn't NumericOpError fall into `std::any`?

```suggestion
        return nonstd::make_unexpected<std::any, NumericOpError>(e.error());
```

Reply:
Yes, you are right. I have made that exact mistake multiple times now